c10/util/BFloat16.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+// Defines the bloat16 type (brain floating-point). This representation uses
+// 1 bit for the sign, 8 bits for the exponent and 7 bits for the mantissa.
+
+#include <c10/macros/Macros.h>
+#include <cmath>
+
+namespace c10 {
+
+namespace detail {
+  inline float f32_from_bits(uint16_t src) {
+    float res = 0;
+    uint16_t* tmp = reinterpret_cast<uint16_t*>(&res);
```

Comment:
Technically, doing this reinterpret cast violates the strict aliasing rule. The portable way to do this conversion is using a `memcpy`; there's some examples of this in `Half.h`